python/graphstorm/model/utils.py

```diff
@@ -311,9 +312,111 @@ def _exchange_node_id_mapping(local_rank, world_size, device,
+    """ Distribute nid_map to all workers.
+
+        Parameters
+        ----------
+        embeddings : DistTensor
+            Embeddings to save
+        local_rank : int
+            Local rank
+        world_size : int
+            World size in a distributed env.
+        node_id_mapping_file: str
+            Path to the file storing node id mapping generated by the
+            graph partition algorithm.
+        device: torch device
+            Device used for all_to_allv data exchange. For gloo backend
+            we store data in CPU, For nccl backend, we need to store
+            data in GPU.
+
+        Returns
+        _______
+        Dict of list: Embeddings index from original order.
+    """
+    assert node_id_mapping_file is not None
+    if isinstance(embeddings, (dgl.distributed.DistTensor, LazyDistTensor)):
+        # only host 0 will load node id mapping from disk
+        if local_rank == 0:
+            ori_node_id_mapping = th.load(node_id_mapping_file)
+            _, node_id_mapping = th.sort(ori_node_id_mapping)
+        else:
+            node_id_mapping = None
+
+        nid_mapping = _exchange_node_id_mapping(
+            local_rank, world_size, device, node_id_mapping, len(embeddings))
+    elif isinstance(embeddings, dict):
+        nid_mapping = {}
+        # only host 0 will load node id mapping from disk
+        node_id_mappings = th.load(node_id_mapping_file) \
+            if local_rank == 0 else None
+
+        for name, emb in embeddings.items():
+            if local_rank == 0:
+                assert name in node_id_mappings, \
+                    f"node id mapping for ntype {name} should exists"
+                # new mapping back index
+                ori_node_id_mapping = node_id_mappings[name]
+                _, node_id_mapping = th.sort(ori_node_id_mapping)
+            else:
+                node_id_mapping = None
+
+            nid_mapping[name] = _exchange_node_id_mapping(
+                local_rank, world_size, device, node_id_mapping, len(emb))
+    else:
+        nid_mapping = None
+    return nid_mapping
+
+def remap_embeddings(embeddings, local_rank, world_size,
+    node_id_mapping_file, device=th.device('cpu')):
+    """ Remap embeddings by nid_map without writing to disk.
+
+        Parameters
+        ----------
+        embeddings : DistTensor
+            Embeddings to save
+        local_rank : int
+            Local rank
+        world_size : int
+            World size in a distributed env.
+        node_id_mapping_file: str
+            Path to the file storing node id mapping generated by the
+            graph partition algorithm.
+        device: torch device
+            Device used for all_to_allv data exchange. For gloo backend
+            we store data in CPU, For nccl backend, we need to store
+            data in GPU.
+
+        Returns
+        _______
+        DistTensor : remapped DistTensor
+    """
+    assert node_id_mapping_file is not None
```

Comment:
You need to add a TODO: handle node_id_mapping_file is None. 
Actually, if node_id_mapping_file is None, We can directly save the embedding in the file.

Reply:
We can directly use the code from L460-476 to handle both node_id_mapping_file is None and node_id_mapping_file is not None.